src/OVAL/oval_state.c

```diff
@@ -219,15 +219,15 @@ void oval_state_set_subtype(struct oval_state *state, oval_subtype_t subtype)
 void oval_state_add_note(struct oval_state *state, char *notes)
 {
 	__attribute__nonnull__(state);
-	oval_collection_add(state->notes, (void *)strdup(notes));
+	oval_collection_add(state->notes, (void *)oscap_strdup(notes));
 }
 
 void oval_state_set_comment(struct oval_state *state, char *comm)
 {
 	__attribute__nonnull__(state);
 	if (state->comment != NULL)
 		free(state->comment);
-	state->comment = comm == NULL ? NULL : strdup(comm);
+	state->comment = comm == NULL ? NULL : oscap_strdup(comm);
```

Comment:
oscap_strdup handles NULL already. you can collapse the logic as I do in the other PR #928